pkg/statsd/statsd.go

```diff
@@ -138,8 +139,12 @@ func (s *Server) RunWithCustomSocket(ctx context.Context, sf SocketFactory) erro
 	}
 
 	bufferSize := 1000 // Estimating this is hard, and tends to cause loss under adverse conditions
-	statser := stats.NewInternalStatser(bufferSize, s.InternalTags, namespace, hostname, metrics, events)
-	// TODO: Make internal metric dispatch configurable
+	var statser stats.Statser
+	if s.DisableInternalMetrics {
```

Comment:
This should be able to select any implementation (specifically the logger should be an option as well).